content/docs/learn/typed-errors/from-either-to-raise.md

```diff
@@ -0,0 +1,181 @@
+to call `bind`. This "injects" any potential error value in the context,
+and continues normal execution otherwise.
+In this style you don't need to use any function to compose computations,
+you use regular Kotlin idioms with some `bind`s sprinkled. In fact
+you could have written the previous code in one single line:
+
+```kotlin
+fun tururu(n: Int): Either<Error, String> = either {
+  g(f(n).bind()).bind().summarize()
+}
+```
+
+How you split your code into local values no longer depends on the
+structure of your functions, as is the case with `flatMap`, but rather
+on the logical decomposition you want in your code.
+
+Arrow provides different builders for different return types
+(`either`, `option`, `result`), but regardless of the one you choose
+you always use `bind` for the injection phase.
+
+## Returning with logical error
+
+Using wrapper types you often have a specific constructor that represents
+the error case. For example, `Left` signals failure when using `Either`.
+Using the `Raise` DSL you no longer need to remember each of them, 
+failure is always signaled using `raise`.
+
+```kotlin
+fun tururuThatRaise(n: Int): Either<Error, String> = either {
+  if (n < 0) raise(Error.NegativeInput)
+  val s = f(n).bind()
+  val t = g(s).bind()
+  t.summarize()
+}
+```
+
+Calling `raise` immediately ends the execution of the current block. In the example
+above, calling `h(-1)` ends up in `raise`, which in the case of an `either` block
+means resulting in `Either.Left(Error.NegativeInput)`.
+
+:::info Somebody said "exceptions"?
+
+The propagation and early return of the `Raise` scope look similar to exceptions.
+However, typed errors should be used for _logical_ errors — problems that have
+a place in your domain model — as opposed to _exceptional_ cases — which represent
+circumstances that are difficult to recover from.
+
+:::
+
+## Transforming the error values
+
+The fact that both `f` and `g` share the same `Error` type is key for
+the previous code blocks to compile. In every `Raise` scope, we have one _single_
+error type which you can `raise` or `bind` in the whole block of code.
+If this is not the case, you have to _bridge_ the different error types, which
+is done using functions like `mapLeft`.
+
+```kotlin
+fun tarara(n: Int): Either<Error, String> =
+  f(n).flatMap { s ->
+    h(s).mapLeft { 
+      it.toString()
+    }.map { t ->
+      t.summarize()
+    }
+  }
+```
+
+Within the `Raise` DSL, the corresponding function is called `withError`.
+Following the discussion above, the `withError` function creates a new
+scope in which the error type is different. The first argument to
+`withError` describes how to transform failure values from the inner
+scope into the error type of the outer scope.
+
+```kotlin
+fun tarara(n: Int): Either<Error, String> = either {
+  val s = f(n).bind()
+  val t = withError({ it.toString() }) { h(s).bind() }
+  t.summarize()
+}
+```
```

Comment:
I'm a bit confused here.  `it.toString()` returns a string, but that's supposed to be the error side.  So it should be an `Error` type, shouldn't it?

Reply:
It would be very helpful for some of these examples to see the equivalent code of what they turn into.  Eg, if we implemented the above example using just boring old if-else statements, what would it look like?  That helps to visualize what's going on under the hood, rather than it just being magic.

Reply:
Fixed the call. But actually, I don't want people to think "under the hood", since the goal here is to learn how to work with the Raise DSL itself.

Reply:
Understandable, but at least for me, understanding what is going on helps me understand how to use the DSL.  It makes it easier for me to "build up" new knowledge from existing knowledge.